cmd/snap/cmd_sandbox_features.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/i18n"
+)
+
+var shortSandboxFeaturesHelp = i18n.G("Print sandbox features available on the system")
+var longSandboxFeaturesHelp = i18n.G(`
+The sandbox command prints tags describing features of individual sandbox
+components used by snapd on a given system.
+`)
+
+type cmdSandboxFeatures struct {
+	Required []string `long:"required" arg-name:"<backend feature>"`
+}
+
+func init() {
+	addDebugCommand("sandbox-features", shortSandboxFeaturesHelp, longSandboxFeaturesHelp, func() flags.Commander {
+		return &cmdSandboxFeatures{}
+	}, map[string]string{
+		"required": i18n.G("Ensure that given backend:feature is available"),
+	}, nil)
+}
+
+func (cmd cmdSandboxFeatures) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	cli := Client()
+	sysInfo, err := cli.SysInfo()
+	if err != nil {
+		return err
+	}
+
+	sandboxFeatures := sysInfo.SandboxFeatures
+
+	if len(cmd.Required) > 0 {
+		avail := make(map[string]bool)
+		for backend := range sandboxFeatures {
+			for _, feature := range sandboxFeatures[backend] {
+				avail[fmt.Sprintf("%s:%s", backend, feature)] = true
+			}
+		}
+		for _, required := range cmd.Required {
+			if !avail[required] {
+				return fmt.Errorf("sandbox feature not available: %q", required)
```

Comment:
Since you can more than one `--required`, I wonder if it would make more sense to return all those that are missing in the error message, rather than stop at the first one., eg: 
`sandbox features not available: foo:bar`
`sandbox features not available: foo:bar magic:thing seccomp:kernel:angry`

Reply:
I don't know yet, I want to use it to experiment. We can always add it. 

To be honest, I never planned on having an error message but I wasn't sure how to return an error code without a message from go-flags ;)

Reply:
`os.Exit(1)` works. Returning an error is nice and polite, but not required.

OTOH os.Exit is not nice to tests; you could instead add an ad hoc error and catch it in errors.go.

Reply:
Nah, this is fine. It will be useful for not just tests but also people :-)